channeld/channeld.c

```diff
@@ -2952,12 +2974,22 @@ static struct amount_sat check_balances(struct peer *peer,
 					   psbt_output_get_amount(psbt, i),
 					   &psbt->outputs[i].unknowns);
 
-	/* Calculate total channel output amount */
+	/* Calculate original channel output amount */
 	if (!amount_msat_add(&funding_amount,
 			     peer->channel->view->owed[LOCAL],
 			     peer->channel->view->owed[REMOTE]))
 		peer_failed_warn(peer->pps, &peer->channel_id,
 				 "Unable to calculate starting channel amount");
+	if (!amount_msat_add(&funding_amount,
+			     funding_amount,
+			     pending_htlcs[TX_INITIATOR]))
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				 "Unable to calculate starting channel amount");
+	if (!amount_msat_add(&funding_amount,
+			     funding_amount,
+			     pending_htlcs[TX_ACCEPTER]))
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				 "Unable to calculate starting channel amount");
```

Comment:
I cried in the last couple of days to understand this logic here, and I am still not sure 100% for what this code does

Can you clarify this operation? and why do we need to look at the pending htlc? we can not just look at the peer->channel->view? to get the balance of the channel. it should contains the on-flight htlc too, right?

Please add a couple of comments in the code to make the review easy to look at, thanks

Reply:
It seems pending HTLC amounts are not included in the channel->view amount as my initial splicing code assumed. I've asked rusty for some clarification here: https://github.com/ElementsProject/lightning/pull/6748#issuecomment-1749624449

Reply:
I am still confused by this change here.

From the following comment https://github.com/ElementsProject/lightning/blob/master/common/initial_channel.h#L19 the `peer->channel->view` contains all the pending change, this mean that it contains also the `pending_htlcs`?

Is the commit outdated?

Can you clarify these operations for readers, please?

Reply:
as far as I understand, `peer->channel->view` only contains the main outputs.

I think there's a semantic confusion here, because the doc on the `view` field talks about "pending" in the sense of whether protocol changes (such as `add_htlc`) are committed with a signature, but the new code here is using "pending" in the sense of "live" or "in-flight" HTLCs.  should probably drop the use of the world pending.

Reply:
Yes, pending means "uncommitted" in that comment.  Should be fixed!